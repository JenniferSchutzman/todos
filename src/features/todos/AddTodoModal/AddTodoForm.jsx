/* eslint-disable object-curly-newline */
import React from 'react';
import PropTypes from 'prop-types';
import { Form } from 'semantic-ui-react';

const AddTodoForm = props => {
  const { due, handleChange, labels, note, title } = props;

  return (
    <Form>
      <Form.Group widths="equal">
        <Form.Input
          id="todo-form_todo"
          label="Todo"
          name="title"
          onChange={handleChange}
          placeholder="Learn React"
          value={title}
        />
      </Form.Group>
      <Form.Group widths="equal">
        <Form.TextArea
          id="todo-form_note"
          label="Note"
          name="note"
          onChange={handleChange}
          placeholder="Check out this link: https://www.example.com"
          value={note}
        />
      </Form.Group>
      <Form.Group widths="equal">
        <Form.Input
          id="todo-form_labels"
          label="Labels"
          name="labels"
          onChange={handleChange}
          value={labels}
        />
        <Form.Input
          id="todo-form_labels"
          label="Due"
          name="due"
          onChange={handleChange}
          type="date"
          value={due}
        />
      </Form.Group>
    </Form>
  );
};

AddTodoForm.propTypes = {
  due: PropTypes.string.isRequired,
  handleChange: PropTypes.func.isRequired,
  labels: PropTypes.string.isRequired,
  note: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
};

AddTodoForm.defaultProps = {};

export default AddTodoForm;
/* eslint-enable object-curly-newline */
