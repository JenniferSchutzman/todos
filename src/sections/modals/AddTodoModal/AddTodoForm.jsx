import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import { Form, Input } from 'semantic-ui-react';

import { addTodo } from '../../../features/todos/todosReducer';

const mapState = () => ({});

const mapDispatch = {
  addTodo,
};

class AddTodoForm extends Component {
  handleClick = () => {
    const { addTodo } = this.props; // eslint-disable-line no-shadow

    addTodo();
  };

  render() {
    return (
      <Form>
        <Form.Group>
          <Form.Field>
            <Input />
          </Form.Field>
        </Form.Group>
      </Form>
    );
  }
}

AddTodoForm.propTypes = {
  addTodo: PropTypes.func.isRequired,
};

AddTodoForm.defaultProps = {};

export default connect(mapState, mapDispatch)(AddTodoForm);
