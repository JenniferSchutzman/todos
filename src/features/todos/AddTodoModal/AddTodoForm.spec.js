/* eslint-disable react/jsx-filename-extension */
import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import AddTodoForm from './AddTodoForm';

describe('<AddTodoForm />', () => {
  describe('render()', () => {
    test('renders the compnent', () => {
      const wrapper = shallow(<AddTodoForm due=" " handleChange={() => {}} labels=" " note=" " title=" " />);

      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });
});
/* eslint-enable react/jsx-filename-extension */
