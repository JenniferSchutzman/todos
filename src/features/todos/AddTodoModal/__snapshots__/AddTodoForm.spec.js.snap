// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddTodoForm /> render() renders the compnent 1`] = `
<Form
  as="form"
>
  <FormGroup
    widths="equal"
  >
    <FormInput
      as={[Function]}
      control={[Function]}
      id="todo-form_todo"
      label="Todo"
      name="title"
      onChange={[Function]}
      placeholder="Learn React"
      value=" "
    />
  </FormGroup>
  <FormGroup
    widths="equal"
  >
    <FormTextArea
      as={[Function]}
      control={[Function]}
      id="todo-form_note"
      label="Note"
      name="note"
      onChange={[Function]}
      placeholder="Check out this link: https://www.example.com"
      value=" "
    />
  </FormGroup>
  <FormGroup
    widths="equal"
  >
    <FormInput
      as={[Function]}
      control={[Function]}
      id="todo-form_labels"
      label="Labels"
      name="labels"
      onChange={[Function]}
      value=" "
    />
    <FormInput
      as={[Function]}
      control={[Function]}
      id="todo-form_labels"
      label="Due"
      name="due"
      onChange={[Function]}
      type="date"
      value=" "
    />
  </FormGroup>
</Form>
`;
